ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs

```diff
@@ -387,6 +388,10 @@ runWith RunNodeArgs{..} encAddrNtN decAddrNtN LowLevelRunNodeArgs{..} =
                 initLedger
                 llrnChainDbArgsDefaults
                 customiseChainDbArgs'
+
+        forM_ (sanityCheckConfig cfg) $ \issue ->
+          traceWith (consensusSanityCheckTracer rnTraceConsensus) issue
```

Comment:
Hmm, I see that these are currently "just" traced. I agree with https://github.com/IntersectMBO/ouroboros-consensus/pull/874/#issuecomment-1887436828 that the check for security parameter consistency should be an exception, not just a warning. In contrast, eg

> Warning if certain (derived) parameters seem "unreasonable", e.g. the ledger snapshot interval.

from #125 seem like a candidate for a warning.

Did you intend this logic to be part of the implementation of `rnTraceConsensus`? Or would it make sense to let `sanityCheckConfig` return a list of either warnings or exceptions, and trace or throw them here, respectively?